python/lsst/obs/fiberspectrograph/spectrum.py

```diff
@@ -0,0 +1,112 @@
+# This file is part of obs_fiberspectrograph
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("FiberSpectrum",)
+
+import numpy as np
+import astropy.io.fits
+import astropy.units as u
+from ._instrument import FiberSpectrograph
+from .data_manager import DataManager
+import lsst.afw.image as afwImage
+from astro_metadata_translator import ObservationInfo
+
+
+class FiberSpectrum:
+    """Define a spectrum from a fiber spectrograph
+
+    Parameters
+    ----------
+    wavelength : `numpy.ndarray`
+        Spectrum wavelength in units provided by the spectrum file.
+    flux: `numpy.ndarray`
+        Spectrum flux.
+    md: `dict`
+        Dictionary of the spectrum headers.
+    detectorId : `int`
+        Optional Detector ID for this data.
+    """
+
+    def __init__(self, wavelength, flux, md=None, detectorId=0):
+        self.wavelength = wavelength
+        self.flux = flux
+        self.md = md
+
+        self.info = ObservationInfo(md)
+        self.detector = FiberSpectrograph().getCamera()[detectorId]
+
+        self.__Mask = afwImage.MaskX(1, 1)
+        self.getPlaneBitMask = self.__Mask.getPlaneBitMask  # ughh, awful Mask API
+        self.mask = np.zeros(flux.shape, dtype=self.__Mask.array.dtype)
+        self.variance = np.zeros_like(flux)
+
+    def getDetector(self):
+        return self.detector
+
+    def getInfo(self):
```

Comment:
Not something you need to do now, but in future type annotations would really help. You should consider whether you need any of the explicitly getter methods since we are moving away from them for Exposure.